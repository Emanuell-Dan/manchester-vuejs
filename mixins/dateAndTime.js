export default {
	methods: {
    getDateAndTime(time) {
      let date = new Date(time);
      return `${date.toDateString().slice(4)} at ${date.toLocaleTimeString().slice(0, -3)}`;
    }
  }
}
