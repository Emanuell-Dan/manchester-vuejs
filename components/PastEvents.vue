<template>
  <div 
    id="past-events"
    class="past-events relative pt-16"
  >
    <h2 class="past-events__heading font-bold text-center text-2xl md:text-4xl mb-4 md:mb-8">
      Past events
    </h2>
    <ul 
      v-if="events.length"
      class="past-events__list flex justify-center flex-wrap"
    >
      <li 
        v-for="event in events"
        :key="event.id"
        class="past-events__item flex"
      >
        <a 
          :href="event.link"
          target="_blank"
          class="past-events__url bg-white flex items-center rounded-lg w-full m-2"
        >
          <div class="past-events__event-info px-4 py-8">
            <p class="past-events__event-date font-medium">{{ getDateAndTime(event.time) }}</p>
            <h3 class="past-events__event-name font-bold py-4">{{ event.name }}</h3>
            <p class="past-events__event-attendees">{{ event.yes_rsvp_count }} attendees</p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</template>

<script>
import dateAndTime from '@/mixins/dateAndTime.js';

export default {
  name: 'PastEvents',
  mixins: [dateAndTime],
  props: {
    events: {
      type: Array,
      default: () => []
    }
  }
};
</script>
