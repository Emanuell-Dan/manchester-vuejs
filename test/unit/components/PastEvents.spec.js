import PastEvents from '@/components/PastEvents.vue';
import { mount } from '@vue/test-utils';
import dateAndTime from '@/mixins/dateAndTime.js';

describe('PastEvents.vue', () => {
	let cmp;

	beforeEach(() => {
		cmp = mount(PastEvents, {
			mixins: [dateAndTime]
		});
	});

	it('should render with expected html structure', () => {
		expect(cmp.element).toMatchSnapshot();
	});

	describe('No past events', () => {
		it('does not render future event card', () => {
			expect(cmp.find('.past-events__list').exists()).toBeFalsy();
		});
	});

	describe('has past events', () => {
		beforeEach(() => {
			cmp.setProps({
				events: [
					{
						id: '001',
						time: 1551983400000,
						name: 'Next Meetup',
						yes_rsvp_count: 5
					},
					{
						id: '002',
						time: 1551983400000,
						name: 'Java Meetup',
						yes_rsvp_count: 9
					}
				]
			});
		});

		it('renders past event card', () => {
			expect(cmp.find('.past-events__list').exists()).toBeTruthy();
		});

		describe('displays correct past event information', () => {
			it('renders past event time', () => {
				expect(cmp.find('.past-events__event-date').text()).toBe('Mar 07 2019 at 18:30');
			});

			it('renders event name', () => {
				expect(cmp.find('.past-events__event-name').text()).toBe('Next Meetup');
			});

			it('renders amount of attendees', () => {
				expect(cmp.find('.past-events__event-attendees').text()).toBe('5 attendees');
			});
		});
	});
});
