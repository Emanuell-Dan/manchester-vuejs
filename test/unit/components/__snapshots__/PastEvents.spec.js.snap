// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastEvents.vue should render with expected html structure 1`] = `
<div
  class="past-events relative pt-16"
  id="past-events"
>
  <h2
    class="past-events__heading font-bold underline text-center text-2xl md:text-4xl mb-4 md:mb-8"
  >
    
    Past events
  
  </h2>
   
  <!---->
</div>
`;
